frontend_lib/src/mention.js

```diff
@@ -0,0 +1,121 @@
+import { v4 as uuidv4 } from 'uuid'
+import i18n from './i18n.js'
+
+const MENTION_ID_PREFIX = 'mention-'
+const MENTION_CLASS = 'mention'
+const MENTION_ME_CLASS = 'mention-me'
+const MENTION_TAG_NAME = 'span'
+const MENTION_REGEX = /(^|\s)@([a-zA-Z0-9\-_]+)($|\s)/
+
+const depthFirstSearchAndMentionAnalysis = childNodesList => {
+  const childNodesListCopy = [...childNodesList]
+  let i = 0
+
+  childNodesListCopy.forEach((node) => {
+    const value = node.nodeValue
+
+    if (node.nodeName === '#text' && value.includes('@')) {
+      const mentionsInThisNode = value.split(/\s/).filter(token => MENTION_REGEX.test(token))
+
+      if (mentionsInThisNode.length > 0) {
+        let mentionIndex = 0
+        let lastMentionIndex = 0
+        const fragment = document.createDocumentFragment()
+        let htmlTagCounter = 0
+
+        mentionsInThisNode.forEach((mention, i) => {
+          mentionIndex = value.indexOf(mention, lastMentionIndex)
+
+          const mentionWithSpan = document.createElement(MENTION_TAG_NAME)
```

Comment:
I think it should be a let here, you change this "const" in the next lines

Reply:
AFAIK the `let` keyword is useful if you want to re-assign the variable, not if you just want to modify its content.

Reply:
Indeed. Using let instead of const there would probably lead to a linting error.

Reply:
In fact it was a linter error which led me to modify this code.